Updates docs for firmware changes for CM5 / Pi500 and other recent firmware changes. (#3935)
raspberrypi/documentation: documentation/asciidoc/computers/raspberry-pi/boot-eeprom.adoc

== Raspberry Pi boot EEPROM

The following Raspberry Pi models use an EEPROM to boot the system:

> * Flagship models since 4

Reviewer: ```suggestion
* Flagship models since Raspberry Pi 4
```

> * Compute Module models since 4 (including Compute Module 4S)
> * Keyboard models since 400

Reviewer: ```suggestion
* Compute Module models since CM4 (including CM4S)
* Keyboard models since Pi 400
```

> * Flagship models since Raspberry Pi 4
> * Compute Module models since Compute Module 4 (including Compute Module 4S)
> * Keyboard models since Raspberry Pi 400

Reviewer: ```suggestion
* Flagship models since 4
* Compute Module models since 4 (including Compute Module 4S)
* Keyboard models since 400
```

== Raspberry Pi boot EEPROM

> Raspberry Pi 5, Raspberry Pi 4, 400, Compute Module 4, and Compute Module 4S computers use an EEPROM to boot the system. All other models of Raspberry Pi computer use the `bootcode.bin` file located in the boot filesystem.

Reviewer 2: I think we generally don't mention unreleased products in our documentation? But this is obviously @nathan-contino 's call to make.
(also, should probably add CM5 to the list after CM4S?)

Author: Too late for that. Although we should generally refer to products as "Raspberry Pi 5 family" or "Raspberry Pi 4 family" rather than tediously enumerating all products everywhere

Reviewer: Fixed via suggestion.

> Raspberry Pi 5, Compute Module 5, Raspberry Pi 500, Raspberry Pi 4, Raspberry Pi 400, Compute Module 4, and Compute Module 4S computers use an EEPROM to boot the system. All other models of Raspberry Pi computer use the `bootcode.bin` file located in the boot filesystem.

Reviewer: ```suggestion
The following Raspberry Pi models use an EEPROM to boot the system:

* Flagship models since Raspberry Pi 4
* Compute Module models since Compute Module 4 (including Compute Module 4S)
* Keyboard models since Raspberry Pi 400

All other models of Raspberry Pi computer use the `bootcode.bin` file located in the boot filesystem.
```
Problem solved 😉

There are multiple ways to update the bootloader of your Raspberry Pi.

> ==== Raspberry Pi 5, Compute Module 5, Raspberry Pi 500, Raspberry Pi 4 and Raspberry Pi 400

Reviewer 3: Is CM4 intentionally missing?

Reviewer 2: A few lines further down it says "N.B. On Compute Module 4 and CM4S the bootrom is unable to load the `recovery.bin` file from EMMC and automatic bootloader updates are not enabled.
The recommended update mechanism is `rpiboot` or via `flashrom` - see `rpi-eeprom-update -h` for more information."

Should that N.B. be turned into a NOTE: block @nathan-contino ?

Reviewer: Fixed!

There are multiple ways to update the bootloader of your Raspberry Pi.

> ==== Flagship models since 4; Compute Modules since 5; Keyboard models since 400

Reviewer: ```suggestion
==== Flagship models since Raspberry Pi 4; Compute Modules since 5; Keyboard models since 400
```

> ==== Raspberry Pi 5, Compute Module 5, Raspberry Pi 500, Raspberry Pi 4 and Raspberry Pi 400

Reviewer 2: Bit of a nitpick, but when listing a collection of different models, would it make sense to always list them in the same order for consistency?
I think my preference would be to list flagship models before compute modules, and within each category list things in the order that they're released? So "Raspberry Pi 4, Raspberry Pi 400, Raspberry Pi 5, Raspberry Pi 500 and Compute Module 5", but YMMV.

Reviewer: ```suggestion
==== Flagship models since 4; Compute Modules since 5; Keyboard models since 400
```
Even more generic, forward-compatible, and terse.

Raspberry Pi OS automatically updates the bootloader for important bug fixes. To manually update the bootloader or change the boot order, use xref:configuration.adoc#raspi-config[raspi-config].

> N.B. On Compute Module 4 and CM4S the bootrom is unable to load the `recovery.bin` file from EMMC and automatic bootloader updates are not enabled.

Reviewer: ```suggestion
NOTE: Compute Module 4 and Compute Module 4S do not support automatic bootloader updates because the bootrom cannot load the `recovery.bin` file from eMMC.
```

> NOTE: Compute Module 4 and Compute Module 4S do not support automatic bootloader updates because the bootrom cannot load the `recovery.bin` file from eMMC.
> The recommended update mechanism is `rpiboot` or via `flashrom` - see `rpi-eeprom-update -h` for more information.

Reviewer: ```suggestion
NOTE: Compute Module 4 and Compute Module 4S do not support automatic bootloader updates because the bootrom cannot load the `recovery.bin` file from eMMC. The recommended update mechanism is `rpiboot` or via `flashrom` - see `rpi-eeprom-update -h` for more information.
```
